packages/language-server/src/completion/completions.ts

```diff
@@ -161,11 +161,20 @@ export function getSuggestionForFieldAttribute(
 
   suggestions.push(...fieldAttributes)
 
-  if (!(currentLine.includes('Int') || currentLine.includes('String'))) {
+  const fieldType = getFieldType(currentLine)
+  if (!fieldType) {
+    return
+  }
+
+  const modelOrEnum = getModelOrTypeOrEnumBlock(fieldType, lines)
+  const isAtIdAllowed = fieldType === 'Int' || fieldType === 'String' || modelOrEnum?.type === 'enum'
```

Comment:
So it's actually ok to not suggest it then 😄

Reply:
That is an interesting fundametnal discussion: Do we want to suggest all the things that validate, or only the things we want to encourage?

Reply:
With `id DateTime @id` `id Float @id` it sounds very bad
Maybe we should still suggest it because a handful of people would actually know how to use that. But then it's a "foot gun" for most people, I guess?

Reply:
added a comment for now

Reply:
Update per discussion on [slack](https://prisma-company.slack.com/archives/C02FNFLDUS3/p1670510516990539?thread_ts=1670510452.456149&cid=C02FNFLDUS3):
It is possible to display hints / tooltips
![image](https://user-images.githubusercontent.com/29753584/206510897-b6272370-b9d2-4cc0-bde0-a611c7010b4b.png)
Therefore, I am of the mind that we enable `@id` for all fields that are already supported in engines. _Then_ we can work on adding these tooltips for usage of `@id` on fields that wouldn't be recommended with an explanation as to why.